### ISSUE TITLE
Mismatch error when loading CLIP-ViT-L/14 to train with SD 1.5

### ISSUE THREAD

**Issue by Exuan148** [tencent-ailab/IP-Adapter#306]
Hello, I know you used CLIP-ViT-H/14 to encode image prompt when training IP-adapter with SD 1.5. I would like to ask how can I use CLIP-ViT-L/14? Since when I loaded CLIP-ViT-L/14 with CLIPVisionModelWithProjection, I encounterd a size mismatch error. It looks like CLIPVisionModelWithProjection can only load CLIP-ViT-H/14? Thank you.

**Comment by Foolbee** [comment]
Hi, I am doing the same trials as you. I just modify the params of the image_proj_model and it did work. But there is one thing I am really concerned that when training with a new image encoder we also need billions of data which is really costy

**Comment by Exuan148** [comment]
Yes, you are right, image_proj_model is only for CLIP contrastive learning, it is unuseful for ip-adapter. What is your task? I think adapter is easy to train when focus on a single field, we have no need to use such amount of data.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tencent-ailab/IP-Adapter"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年5月25日(星期六) 中午1:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tencent-ailab/IP-Adapter] Mismatch error when loading CLIP-ViT-L/14 to train with SD 1.5 (Issue #306)





 
Hi, I am doing the same trials as you. I just modify the params of the image_proj_model and it did work. But there is one thing I am really concerned that when training with a new image encoder we also need billions of data which is really costy

**Comment by Foolbee** [comment]
From my point of view, the adapter module also need enough image data to function well, and that's why the author use a 5B dataset to train the adapter, I am consider using other modalities to finetune the adapter but there is no pretrained adapter model using CLIP-Large-patch4, so i have to consider training from scratch(from the origin attention processor)
